app/src/main/java/com/infomaniak/mail/ui/bottomSheetDialogs/ScheduleSendBottomSheetDialog.kt

```diff
@@ -0,0 +1,224 @@
+
+        localSettings.lastSelectedScheduleDate?.let { lastSelectedSchedule ->
+            if (Date(lastSelectedSchedule).isAtLeastXMinutesInTheFuture(MIN_SCHEDULE_DELAY_MINUTES)) {
+                lastScheduleItem.isVisible = true
+                lastScheduleItem.setDescription(
+                    mostDetailedDate(
+                        context,
+                        date = Date(lastSelectedSchedule),
+                        format = FORMAT_DATE_DAY_MONTH,
+                    )
+                )
+            }
+        }
+
+        lastScheduleItem.setClosingOnClickListener {
+            val draftResource = navigationArgs.draftResource
+            val lastSelectedScheduleDate = localSettings.lastSelectedScheduleDate
+
+            if (navigationArgs.isAlreadyScheduled) {
+                if (draftResource != null && lastSelectedScheduleDate != null) {
+                    trackScheduleSendEvent("lastSchedule")
+                    mainViewModel.rescheduleDraft(draftResource, Date(lastSelectedScheduleDate))
+                }
+            } else {
+                lastSelectedScheduleDate?.let {
+                    trackScheduleSendEvent("lastSchedule")
+                    newMessageViewModel.setScheduleDate(Date(it))
+                    newMessageViewModel.triggerSendMessage()
+                }
+            }
+        }
+
+        if (newMessageViewModel.currentMailbox.isFree) {
+            customScheduleItem.setOnClickListener {
+                safeNavigate(
+                    resId = R.id.upgradeProductBottomSheetDialog,
+                    currentClassName = ScheduleSendBottomSheetDialog::class.java.name,
+                )
+            }
+        } else {
+            customScheduleItem.setClosingOnClickListener {
+                if (navigationArgs.isAlreadyScheduled) {
+                    mainViewModel.showSelectDateAndTimeForScheduleDialog()
+                } else {
+                    newMessageViewModel.showSelectDateAndTimeForScheduleDialog()
+                }
+            }
+        }
+
+        fun createScheduleItem(schedule: Schedule): ActionItemView = ActionItemView(this.context).apply {
+            setTitle(schedule.scheduleTitleRes)
+            setDescription(mostDetailedDate(context, date = schedule.date, format = FORMAT_DATE_DAY_MONTH))
+            setIconResource(schedule.scheduleIconRes)
+            setClosingOnClickListener {
+                if (navigationArgs.isAlreadyScheduled) {
+                    navigationArgs.draftResource?.let {
+                        trackScheduleSendEvent(schedule.matomoValue)
+                        mainViewModel.rescheduleDraft(draftResource = it, schedule.date)
+                    }
+                } else {
+                    trackScheduleSendEvent(schedule.matomoValue)
+                    newMessageViewModel.setScheduleDate(schedule.date)
+                    newMessageViewModel.triggerSendMessage()
+                }
+            }
+        }
```

Comment:
This method doesn't need to be an inner method, it makes the onViewCreated heavier and harder to read for no reason